### ISSUE TITLE
Hebrew locale not work in Sparkle

### ISSUE THREAD

**Issue by uiryuu** [iina/iina#4938]
**System and IINA version:**

- macOS macOS 14.4.1
- IINA `enable-he` branch

**Expected behavior:**
Sparkle is running in Hebrew

**Actual behavior:**
<img width="549" alt="image" src="https://github.com/iina/iina/assets/20237141/9b2c7a09-6567-4e38-8c1f-fb827f4ddf5c">

**Steps to reproduce:**
- Start IINA for the first time in Hebrew

It seems the sparkle has support for Hebrew locale (https://github.com/sparkle-project/Sparkle/tree/2.x/Sparkle/he.lproj). Didn't spent too much time into this. Just created this issue as a reminder.

- [x] MPV does not have this problem.

**How often does this happen?**
Should be always

**Comment by ShlomoCode** [comment]
This seems to be intentional behavior from Sparkle, they use the MacOS language and not the language defined for the app
https://github.com/sparkle-project/Sparkle/issues/1083#issuecomment-305223877

**Comment by uiryuu** [comment]
No, by "provided by macOS" they meant the language option provided by the macOS at runtime. Xcode will inject this option when we intend to run the app in a different language than the system language. The root cause of this is that the Hebrew localization is not completed, so untranslated strings can be only shown in English. See this

![image](https://github.com/iina/iina/assets/20237141/9f7db89c-c96a-4d90-bfd9-2d2ff37e213f)

Clearly it knows that we are running in Hebrew, but just lacking some Hebrew translations. I'm labeling upstream.

**Comment by ShlomoCode** [comment]
> but just lacking some Hebrew translations. I'm labeling upstream.

https://github.com/sparkle-project/Sparkle/pull/2573

**Comment by ShlomoCode** [comment]
@uiryuu Sparkle [v2.7.0](https://github.com/sparkle-project/Sparkle/releases/tag/2.7.0) has just been released and includes my Hebrew translations
